### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by silviatofana**
In this project, you will connect your React-Redux application to an existing API to create and remove books on a remote server.

IMPORTANT NOTE: Read all requirements before you start building your project.

General requirements
Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
Make sure that you used correctly [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
JavaScript requirements
Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
Project requirements
In the previous project, you developed the add book and remove book features directly in your application using Redux and React state. Now you should refactor your code to use the Bookstore API to add and remove books.

Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
Fetch book list from Bookstore API on Books page load.
Add an Action Creator, that returns a function.
Fetch a list of books.
Dispatch another action (add an action type) that will pass the API response to the reducer and update the state.
Refactor your add book and remove book features to persist your changes in the server.
Change your existing Action Creators, so they return functions.
Make changes by using API.
Dispatch an action (action type that existed before refactor) that will update the state accordingly.
Add a book and remove book should work in the same way after the refactor.
No styling is required.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
     "react-test-renderer": "^18.2.0",
+    "redux": "^4.2.0",
+    "redux-thunk": "^2.4.1",
```
Comment (youmari) [inline package.json:19]: - [ ] When I run your application I got three error of dependencies but don't exist in the `package.json` file like `axios` and react-circular please try to fix that 
image

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @silviatofana , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
The repo looks professional.✔️
There are no linter errors✔️
You have descriptive PR✔️
Your project looks awesome✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @silviatofana        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by silviatofana** [comment]
thanks

**Pull request merged**